### ISSUE TITLE
[ignition-rendering] cmake configuration fails

### ISSUE THREAD

**Issue by doctorcolossus** [acxz/gazebo-arch#41]
Trying to install `ignition-gazebo` fails for me while attempting to build dependency `ignition-rendering`.

There are several warnings, but it isn't clear for me what the error is.

I have an onboard Intel GPU, so no CUDA. It should still work without that, right?

Can anyone help? I need to get this installed urgently.

```
$ yay ignition-gazebo
1 aur/ignition-gazebo 6.12.0-1 (+0 0.00)
    Open source robotics simulator.
==> Packages to install (eg: 1 2 3, 1-3 or ^4)
==> 1
:: Checking for conflicts...
:: Checking for inner conflicts...
[Aur:7]  ignition-rendering-6.5.0-1  ignition-msgs-8.6.0-1  ignition-transport-11.2.0-1  ignition-gui-6.6.1-1  ignition-fuel_tools-7.1.0-1  ignition-sensors-6.6.0-1  ignition-gazebo-6.12.0-1

  7 ignition-rendering                       (Build Files Exist)
  6 ignition-msgs                            (Build Files Exist)
  5 ignition-transport                       (Build Files Exist)
  4 ignition-gui                             (Build Files Exist)
  3 ignition-fuel_tools                      (Build Files Exist)
  2 ignition-sensors                         (Build Files Exist)
  1 ignition-gazebo                          (Build Files Exist)
==> Packages to cleanBuild?
==> [N]one [A]ll [Ab]ort [I]nstalled [No]tInstalled or (1 2 3, 1-3, ^4)
==> a
:: Deleting (1/7): ~/.cache/yay/ignition-rendering
:: Deleting (2/7): ~/.cache/yay/ignition-msgs
:: Deleting (3/7): ~/.cache/yay/ignition-transport
:: Deleting (4/7): ~/.cache/yay/ignition-gui
:: Deleting (5/7): ~/.cache/yay/ignition-fuel_tools
:: Deleting (6/7): ~/.cache/yay/ignition-sensors
:: Deleting (7/7): ~/.cache/yay/ignition-gazebo
:: (1/7) Downloaded PKGBUILD: ignition-gui
:: (2/7) Downloaded PKGBUILD: ignition-transport
:: (3/7) Downloaded PKGBUILD: ignition-msgs
:: (4/7) Downloaded PKGBUILD: ignition-fuel_tools
:: (5/7) Downloaded PKGBUILD: ignition-sensors
:: (6/7) Downloaded PKGBUILD: ignition-gazebo
:: (7/7) Downloaded PKGBUILD: ignition-rendering
  7 ignition-rendering                       (Build Files Exist)
  6 ignition-msgs                            (Build Files Exist)
  5 ignition-transport                       (Build Files Exist)
  4 ignition-gui                             (Build Files Exist)
  3 ignition-fuel_tools                      (Build Files Exist)
  2 ignition-sensors                         (Build Files Exist)
  1 ignition-gazebo                          (Build Files Exist)
==> Diffs to show?
==> [N]one [A]ll [Ab]ort [I]nstalled [No]tInstalled or (1 2 3, 1-3, ^4)
==> n
:: (1/7) Parsing SRCINFO: ignition-rendering
:: (2/7) Parsing SRCINFO: ignition-msgs
:: (3/7) Parsing SRCINFO: ignition-transport
:: (4/7) Parsing SRCINFO: ignition-gui
:: (5/7) Parsing SRCINFO: ignition-fuel_tools
:: (6/7) Parsing SRCINFO: ignition-sensors
:: (7/7) Parsing SRCINFO: ignition-gazebo
==> Making package: ignition-gui 6.6.1-1 (2022 m. spalio 13 d. 22:11:50)
==> Retrieving sources...
==> Making package: ignition-sensors 6.6.0-1 (2022 m. spalio 13 d. 22:11:50)
==> Retrieving sources...
  -> Downloading ignition-gui-6.6.1.tar.gz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  -> Downloading ignition-sensors-6.6.0.tar.gz...
==> Making package: ignition-gazebo 6.12.0-1 (2022 m. spalio 13 d. 22:11:50)
==> Making package: ignition-rendering 6.5.0-1 (2022 m. spalio 13 d. 22:11:50)
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
 ==> Retrieving sources...
                                Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0==> Retrieving sources...
  -> Downloading ignition-gazebo-6.12.0.tar.gz...
  -> Downloading ignition-rendering-6.5.0.tar.gz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
==> Making package: ignition-msgs 8.6.0-1 (2022 m. spalio 13 d. 22:11:51)
==> Making package: ignition-transport 11.2.0-1 (2022 m. spalio 13 d. 22:11:51)
==> Retrieving sources...
==> Retrieving sources...
  -> Downloading ignition-msgs-8.6.0...
==> Making package: ignition-fuel_tools 7.1.0-1 (2022 m. spalio 13 d. 22:11:51)
  -> Downloading ignition-transport-11.2.0.tar.gz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Curre==> Retrieving sources...
nt
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  -> Downloading ignition-fuel_tools-7.1.0.tar.gz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0 2 7233k2k      0 --:--:-- --:--:-- --:--:--  232k
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 1192k    0 1192k    0     0   769k      0 --:--:--  0:00:01 --:--:-- 1814k
==> Validating source files with sha256sums...
    ignition-gui-6.6.1.tar.gz ... Passed
100 4117k    0 4117k    0     0  1444k      0 --:--:--  0:00:02 --:--:-- 2020k 6945k
==> Validating source files with sha256sums...
    ignition-sensors-6.6.0.tar.gz ... Passed
100  345k    0  345k    0     0  92511      0 --:--:--  0:00:03 --:--:--  153k   0 --:--:--  0:00:03 --:--:-- 2781k
==> Validating source files with sha256sums...
    ignition-fuel_tools-7.1.0.tar.gz ... Passed
100  344k    0  344k    0     0  91482      0 --:--:--  0:00:03 --:--:--  101k
==> Validating source files with sha256sums...
    ignition-msgs-8.6.0 ... Passed
100  590k    0  590k    0     0   142k      0 --:--:--  0:00:04 --:--:--  220k
==> Validating source files with sha256sums...
    ignition-transport-11.2.0.tar.gz ... Passed
100 15.5M    0 15.5M    0     0  2768k      0 --:--:--  0:00:05 --:--:-- 3249k
==> Validating source files with sha256sums...
    ignition-gazebo-6.12.0.tar.gz ... Passed
100 39.7M    0 39.7M    0     0  6622k      0 --:--:--  0:00:06 --:--:-- 7420k
==> Validating source files with sha256sums...
    ignition-rendering-6.5.0.tar.gz ... Passed
==> Making package: ignition-rendering 6.5.0-1 (2022 m. spalio 13 d. 22:11:57)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Found ignition-rendering-6.5.0.tar.gz
==> Validating source files with sha256sums...
    ignition-rendering-6.5.0.tar.gz ... Passed
==> Removing existing $srcdir/ directory...
==> Extracting sources...
  -> Extracting ignition-rendering-6.5.0.tar.gz with bsdtar
==> Sources are ready.
==> Making package: ignition-rendering 6.5.0-1 (2022 m. spalio 13 d. 22:11:59)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> WARNING: Using existing $srcdir/ tree
==> Starting build()...
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- ignition-rendering6 version 6.5.0
-- Operating system is Linux
-- Looking for ignition-math6 -- found version 6.13.0
-- Searching for dependencies of ignition-math6
-- Searching for <ignition-math6> component [eigen3]
-- Looking for ignition-math6-eigen3 -- found version 6.13.0
-- Searching for dependencies of ignition-math6-eigen3
-- Looking for ignition-math6 - found

CMake Warning at /usr/share/cmake/ignition-cmake2/cmake2/IgnUtils.cmake:197 (find_package):
  By not providing "Findignition-common4.cmake" in CMAKE_MODULE_PATH this
  project has asked CMake to find a package configuration file provided by
  "ignition-common4", but CMake did not find one.

  Could not find a package configuration file provided by "ignition-common4"
  with any of the following names:

    ignition-common4Config.cmake
    ignition-common4-config.cmake

  Add the installation prefix of "ignition-common4" to CMAKE_PREFIX_PATH or
  set "ignition-common4_DIR" to a directory containing one of the above
  files.  If "ignition-common4" provides a separate development package or
  SDK, be sure it has been installed.
Call Stack (most recent call first):
  CMakeLists.txt:38 (ign_find_package)


-- Looking for ignition-common4 - not found

-- Looking for ignition-plugin1 -- found version 1.3.0
-- Searching for dependencies of ignition-plugin1
-- Searching for <ignition-plugin1> component [all]
-- Looking for all libraries of ignition-plugin1 -- found version 1.3.0
-- Looking for ignition-plugin1 -- found version 1.3.0
-- Looking for ignition-plugin1-loader -- found version 1.3.0
-- Searching for dependencies of ignition-plugin1-loader
-- Looking for ignition-plugin1-register -- found version 1.3.0
-- Searching for dependencies of ignition-plugin1-register
-- Looking for ignition-plugin1 - found

-- Found FreeImage: TRUE (Required is at least version "3.9")
-- Looking for FreeImage - found

-- Found OpenGL: /usr/lib/libOpenGL.so
-- Looking for OpenGL - found

-- Checking for module 'OGRE-RTShaderSystem >= 1.9'
--   Found OGRE-RTShaderSystem , version 1.9.0
-- Checking for module 'OGRE-Terrain >= 1.9'
--   Found OGRE-Terrain , version 1.9.0
-- Checking for module 'OGRE-Overlay >= 1.9'
--   Found OGRE-Overlay , version 1.9.0
-- Checking for module 'OGRE-Paging >= 1.9'
--   Found OGRE-Paging , version 1.9.0
-- Looking for IgnOGRE - found

-- Found Boost: /usr/lib64/cmake/Boost-1.80.0/BoostConfig.cmake (found version "1.80.0")
-- Looking for OGRE using the name: OGRE2
--   ! OGRE2 not found
-- Looking for OGRE using the name: OGRE-Next
--   ! OGRE-Next not found
-- Looking for IgnOGRE2 - not found

CUDA_TOOLKIT_ROOT_DIR not found or specified
-- Could NOT find CUDA (missing: CUDA_TOOLKIT_ROOT_DIR CUDA_NVCC_EXECUTABLE CUDA_INCLUDE_DIRS CUDA_CUDART_LIBRARY)
-- optix library not found.  Please locate before proceeding.
-- OptiX headers (optix.h and friends) not found.  Please locate before proceeding.
-- optix Prime library not found.  Please locate before proceeding.
-- Could NOT find OptiX (missing: OptiX_FOUND) (Required is at least version "3.8.0")
-- Looking for OptiX - not found

CMake Warning at /usr/share/cmake/ignition-cmake2/cmake2/IgnConfigureBuild.cmake:55 (message):
   CONFIGURATION WARNINGS:
   -- Skipping component [optix]: Missing dependency [OptiX].
      ^~~~~ Set SKIP_optix=true in cmake to suppress this warning.

Call Stack (most recent call first):
  CMakeLists.txt:170 (ign_configure_build)


CMake Error at /usr/share/cmake/ignition-cmake2/cmake2/IgnConfigureBuild.cmake:59 (message):
  -- BUILD ERRORS: These must be resolved before compiling.
Call Stack (most recent call first):
  CMakeLists.txt:170 (ign_configure_build)


CMake Error at /usr/share/cmake/ignition-cmake2/cmake2/IgnConfigureBuild.cmake:61 (message):
  -- 	Missing dependency [ignition-common4] (Components: graphics, events)
Call Stack (most recent call first):
  CMakeLists.txt:170 (ign_configure_build)


CMake Error at /usr/share/cmake/ignition-cmake2/cmake2/IgnConfigureBuild.cmake:63 (message):
  -- END BUILD ERRORS

Call Stack (most recent call first):
  CMakeLists.txt:170 (ign_configure_build)


CMake Error at /usr/share/cmake/ignition-cmake2/cmake2/IgnConfigureBuild.cmake:68 (message):
  Errors encountered in build.  Please see BUILD ERRORS above.
Call Stack (most recent call first):
  CMakeLists.txt:170 (ign_configure_build)


-- Configuring incomplete, errors occurred!
See also "~/.cache/yay/ignition-rendering/src/gz-rendering-ignition-rendering6_6.5.0/build/CMakeFiles/CMakeOutput.log".
==> ERROR: A failure occurred in build().
    Aborting...
 -> error making: ignition-rendering
```

**Comment by acxz** [comment]
> CMake Error at /usr/share/cmake/ignition-cmake2/cmake2/IgnConfigureBuild.cmake:61 (message):
  -- 	Missing dependency [ignition-common4] (Components: graphics, events)

What's your installed `ignition-common` version?
`yay -Qs ignition-common`
If you don't have it installed, try `yay -S ignition-common`. You may have to remove `ignition-common-3` via `yay -Rns ignition-common-3`.

**Comment by doctorcolossus** [comment]
Thanks for your quick reply.

I had `ignition-common-3` (from arch4edu I think) installed during my first few attempts, then I uninstalled it, seeing that `ignition-common` was listed as the main dependency with `ignition-common-3` as an alternative, and assuming that the former would get installed automatically as a dependency, but actually I don't see that happening and now I apparently have neither installed:
```
$  yay -Qs ignition-common
local/ignition-common-3 3.14.1-1
    A collection of useful classes and functions for handling many command tasks. This includes parsing 3D mesh files, managing console
    output, and using PID controllers.
```
Let me try manually pre-installing that, and then trying again to build `ignition-rendering`...

Okay, so I installed `ignition-common` (4.5.2-1) and it now `ignition-rendering` does configure correctly and is currently building...

Hopefully it succeeds and then I can get the whole `ignition-gazebo` installed, but I guess that resolves this issue! Not sure why `yay` didn't try to resolve that dependency though... 🤔 Thanks again for the quick help! If I run into any more problems, I'll open a new issue.

**Comment by doctorcolossus** [comment]
Maybe `ignition-common-3` should not be listed as an alternative though.

I'm pretty sure my cmake configuration failed similarly when I tried with that one.

**Comment by acxz** [comment]
Yeah the proper solution to this would be to specify `ignition-common>=4` in the `makedepends` for `ignition-gazebo`. Feel free to submit a PR about it, in the meanwhile I'll keep this issue open so its more visible if others encounter this issue.

**Comment by goekce** [comment]
I had a similar error:
```
Missing dependency [ignition-common5] (Components: graphics, ...
```
Problem: `gdal` and `assimp` were missing from `ignition-common` so graphics components were not built.

**Comment by acxz** [comment]
This should be resolved with 38a7924
